src/os/ui/menu/layermenu.js

```diff
@@ -367,11 +367,29 @@ os.ui.menu.layer.onFeatureList_ = function(event) {
  * @private
  */
 os.ui.menu.layer.onGoTo_ = function(event) {
-  var extent = os.ui.menu.layer.getLayersFromContext(event.getContext())
-      .reduce(os.fn.reduceExtentFromLayers, ol.extent.createEmpty());
+  // aggregate the features and execute os.feature.flyTo, in case they have altitude and pure extent wont cut it
+  const layers = os.ui.menu.layer.getLayersFromContext(event.getContext());
+  const features = layers.reduce((feats, layer) => {
+    let source = layer.getSource();
+    if (source instanceof os.source.Request) {
+      source = /** @type {os.source.Request} */ (source);
+      return feats.concat(
+        source.getFeatureCount() > 0 ?
```

Comment:
This branch is a little bit cleaner little cleaner both speed and memory-wise if you do:
```javascript
return source.getFeatureCount() ?
  feats.concat(source.getFeatures()) :
  feats;
```
since `feats.concat([])` is effectively `feats.slice()`; i.e. copies to a new array.